### PR TITLE
Add missing arch-specific dependencies to tf_to_kernel

### LINKED ISSUE #45104: Missing dependency on PowerPCCodeGen leads to undefined references in tf_to_kernel
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): RHEL 7.4, PPC

**Describe the problem**

Running the tests with `bazel test ...` leads to undefined reference errors, see below log extract.
The reason is that `InitializeNativeTarget` is called which seemingly is a generated function and while the x86 dependency lib is linked via `@llvm-project//llvm:X86CodeGen`, the PPC equivalent is not.

The same problem may also affect ARM and arch64. Compare https://github.com/tensorflow/tensorflow/blob/15f4bda049539dd41c6dd9d0737d33da86cc32cf/tensorflow/compiler/aot/BUILD#L73-L83
with https://github.com/tensorflow/tensorflow/blob/15f4bda049539dd41c6dd9d0737d33da86cc32cf/tensorflow/compiler/mlir/tools/kernel_gen/BUILD#L122-L123

I think those lists should be pretty much identical.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

`bazel test //tensorflow/...`

**Any other info / logs**

```
SUBCOMMAND: # //tensorflow/core/kernels/mlir_generated:invert_i32_kernel_cubin [action 'compile tensorflow/core/kernels/mlir_generated/invert_i32_kernel_cubin.sm_37.bin', configuration: 18b45a5a4afc1ae33c4d05
7c86644439e6ecd019473065d664167c1910aff23b, execution platform: @local_execution_config_platform//:platform]
...
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCTargetInfo'
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCTarget'
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCTargetMC'
bazel-out/ppc-opt/bin/tensorflow/compiler/mlir/tools/kernel_gen/_objs/tf_to_kernel/tf_to_kernel.o:tf_to_kernel.cc:function main: error: undefined reference to 'LLVMInitializePowerPCAsmPrinter'
collect2: error: ld returned 1 exit status
```

### PR DESCRIPTION
**Pull request by Flamefire**
The tf_to_kernel target is missing some dependencies which shows on e.g. POWER by undefined reference errors
Fixes #45104

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by rthadur** [comment]
@mihaimaruseac can you please help with pulling this PR in ?

**Comment by mihaimaruseac** [comment]
I'm worried that so many builds are failing. Can we first rebase this on master and trigger a new test run first?

**Comment by Flamefire** [comment]
@mihaimaruseac rebased without any changes on current master.

File: tensorflow/compiler/mlir/tools/kernel_gen/BUILD [hunk 1]
```diff
@@ -135,6 +138,11 @@ tf_cc_binary(
         "@llvm-project//mlir:Pass",
         "@llvm-project//mlir:TargetLLVMIR",
     ],
+    ] + if_llvm_system_z_available([
```
Comment (mihaimaruseac) [inline tensorflow/compiler/mlir/tools/kernel_gen/BUILD:141]: This causes a syntax error. I think you want to remove line 140 (GitHub doesn't let me do an automatic suggestion and merge)

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/mlir/tools/kernel_gen/BUILD [hunk 2]
```diff
@@ -135,6 +138,11 @@ tf_cc_binary(
         "@llvm-project//mlir:Pass",
         "@llvm-project//mlir:TargetLLVMIR",
     ],
```
Comment (Flamefire) [inline tensorflow/compiler/mlir/tools/kernel_gen/BUILD:140]:
```suggestion
```

File: tensorflow/compiler/mlir/tools/kernel_gen/BUILD (hunk 1 above)
Comment (Flamefire) [reply tensorflow/compiler/mlir/tools/kernel_gen/BUILD:140]: Good catch. Not sure where that was missed, maybe by the rebase. Anyway, fixed

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**